src/java/org/apache/cassandra/service/accord/AccordVerbHandler.java

```diff
@@ -27,22 +27,32 @@
 import accord.messages.Request;
 import org.apache.cassandra.net.IVerbHandler;
 import org.apache.cassandra.net.Message;
+import org.apache.cassandra.tcm.ClusterMetadataService;
 
 public class AccordVerbHandler<T extends Request> implements IVerbHandler<T>
 {
     private static final Logger logger = LoggerFactory.getLogger(AccordVerbHandler.class);
 
     private final Node node;
+    private final AccordEndpointMapper configService;
 
-    public AccordVerbHandler(Node node)
+    public AccordVerbHandler(Node node, AccordEndpointMapper endpointMapper)
     {
         this.node = node;
+        this.configService = endpointMapper;
     }
 
     @Override
     public void doVerb(Message<T> message) throws IOException
     {
+        ClusterMetadataService.instance().maybeCatchup(message.epoch());
         logger.debug("Receiving {} from {}", message.payload, message.from());
-        message.payload.process(node, EndpointMapping.getId(message.from()), message);
+        Request request = message.payload;
+        Node.Id from = configService.mappedId(message.from());
+        // CMS updates received in another messaging thread can race with node upates
```

Comment:
this isn't 100% matching `Node.recieve` as that has a `from` and `wait` epoch... in my patch I change to mimic that API, you will see during the rebase =D